rebuild/src/secret_key.cc

```diff
@@ -0,0 +1,171 @@
+#include "secret_key.h"
+
+void SecretKey::Init(
+    Napi::Env env, Napi::Object &exports, BlstTsAddon *module) {
+    Napi::HandleScope scope(env);
+    auto proto = {
+        StaticMethod(
+            "fromKeygen",
+            &SecretKey::FromKeygen,
+            static_cast<napi_property_attributes>(
+                napi_static | napi_enumerable)),
+        StaticMethod(
+            "deserialize",
+            &SecretKey::Deserialize,
+            static_cast<napi_property_attributes>(
+                napi_static | napi_enumerable)),
+        InstanceMethod(
+            "serialize",
+            &SecretKey::Serialize,
+            static_cast<napi_property_attributes>(napi_enumerable)),
+        InstanceMethod(
+            "toPublicKey",
+            &SecretKey::ToPublicKey,
+            static_cast<napi_property_attributes>(napi_enumerable)),
+        InstanceMethod(
+            "sign",
+            &SecretKey::Sign,
+            static_cast<napi_property_attributes>(napi_enumerable)),
+    };
+
+    Napi::Function ctr = DefineClass(env, "SecretKey", proto, module);
+    module->_secret_key_ctr = Napi::Persistent(ctr);
+    exports.Set(Napi::String::New(env, "SecretKey"), ctr);
+
+    exports.Get("BLST_CONSTANTS")
+        .As<Napi::Object>()
+        .Set(
+            Napi::String::New(env, "SECRET_KEY_LENGTH"),
+            Napi::Number::New(env, BLST_TS_SECRET_KEY_LENGTH));
+}
+
+Napi::Value SecretKey::FromKeygen(const Napi::CallbackInfo &info) {
+    BLST_TS_FUNCTION_PREAMBLE(info, env, module)
+    Napi::Value ikm_value = info[0];
+
+    BLST_TS_UNWRAP_UINT_8_ARRAY(ikm_value, ikm, "ikm")
+    // Check for less than 32 bytes so consumers don't accidentally create
+    // zero keys 
+    if (ikm.ByteLength() < BLST_TS_SECRET_KEY_LENGTH) {
+        std::ostringstream msg;
+        msg << "ikm must be greater than or equal to "
+            << BLST_TS_SECRET_KEY_LENGTH << " bytes";
```

Comment:
If possible, I'd avoid using `ostringstream`. Unless something changed recently, they are terribly slow and if you don't need any fancy modifiers (and it seems you only need to concatenate strings), direct string additions will most likely produce more performant code. With `using namespace std::string_literals;` you could have "blabla: "s + std::to_string(42) + "."s. I might be wrong, my C++ is a little bit Rusty. :crab:

Reply:
updated in 5de98fa9dd7e22485d29362dd4da078c14de9093